_posts/2019-12-08-what-is-a-shell.md

```diff
@@ -0,0 +1,166 @@
+For example `man zip`.
+Similar functionality can be achieved in Powershell by using `Get-Command`.
+
+### Manipulating Data
+
+In the previous section we saw how a user may list the available commands or executables they can run via their shell.
+If you've been trying these on your own system,
+you may have noticed an overwhelming amount of output.
+Sometimes you may need all of it,
+but often you might want a subset of the data.
+
+We'll continue with `ls`,
+as it is fairly simple to work with.
+In Bash, if you want to `ls` only directories,
+or in other words, _filter entries by the type directory_,
+then you may use the command `ls -d */`.
+The logic behind this is that the `-d` flag will output the / character after a directory and then we're using a glob to match anything that ends with the /.
+If you're in powershell, you can take advantage of the `-Directory` flag.
+The full command will look like `ls -Directory`.
+
+These shells differ in their approach to how data is handled.
+Bash handles data like raw text while Powershell seems to handle the data as objects.
+Both approaches have advantages and disadvantages,
+however if you need to context switch between shells this can be a lot of mental overhead.
+
+Sometimes simple data manipulation goes well beyond one command.
+If you are experienced in your shell of choice,
+you've probably _piped_ commands together or even written scripts.
+The idea is that commands can take some input, maybe from the user, maybe from something more formalized like standard-in,
+and then pass it into a program.
+Shell programs often output data via standard-out allowing you to continue this pattern of linking programs together.
+
+
+### The Future of Shells
+Many of the popular shells have reached maturity and are industry standards, but like all software, they have their drawbacks.
+I think a fair number of developers probably have to work across multiple operating systems whether their position requires it or they enjoy using a different OS for personal use outside of work.
+Having to maintain multiple shell environments can be tedious and error prone even if it is fun to learn the internals of all of them.
+A recent project, [Nushell](https://www.nushell.sh/) aims to solve some of these issues.
+Nushell is a cross-platform shell written in Rust which works on Windows, MacOS, and Linux.
+It is still in the very early stages and only a few months old as of the time of this writing,
+although development seems to be going at a rapid pace with a 3-week release cycle.
+Aside from the cross-platform aspect of Nushell,
+one of the more appealing features is how Nushell treats data.
+Instead of treating all data as raw text,
+Nushell takes some inspiration from Powershell and treats data as structured tables or objects.
+This enables Nushell to support a rich command and plugin system to manipulate data in very useful and declarative ways.
```

Comment:
Might be nice to link to the installation instructions somewhere in here, or in the next section: https://book.nushell.sh/en/installation